src/Storages/MergeTree/MergeTreeIndexFullText.cpp

```diff
@@ -548,12 +548,21 @@ bool MergeTreeConditionFullText::traverseTreeEquals(
         token_extractor->stringLikeToBloomFilter(value.data(), value.size(), *out.bloom_filter);
         return true;
     }
-    else if (function_name == "hasToken")
+    else if (function_name.starts_with("hasToken"))
     {
         out.key_column = key_column_num;
         out.function = RPNElement::FUNCTION_EQUALS;
         out.bloom_filter = std::make_unique<BloomFilter>(params);
-        const auto & value = const_value.get<String>();
+        const auto value = std::invoke(
+            [&const_value, &function_name]
+            {
```

Comment:
Please add a short comment to the code that the trick works based on an index build from lowercase'd data.